engine/runtime/src/main/java/org/enso/interpreter/node/typecheck/AbstractTypeCheckNode.java

```diff
@@ -0,0 +1,93 @@
+package org.enso.interpreter.node.typecheck;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.Node;
+import java.util.List;
+import org.enso.interpreter.node.ExpressionNode;
+import org.enso.interpreter.node.expression.builtin.meta.AtomWithAHoleNode;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.DataflowError;
+
+/**
+ * Root of hierarchy of nodes checking types. This class (and its subclasses) are an implementation
+ * detail. The API to perform the is in {@link TypeCheckNode}.
```

Comment:
Looks like some typo in the doc - the API to perform the _what_?

Reply:
Given the integration problems I am facing since Friday, I probably integrate and only then fix the comment in subsequent PR.

Reply:
Fair enough